rpcs3/Emu/RSX/Program/ProgramStateCache.cpp

```diff
@@ -425,31 +348,41 @@ vertex_program_utils::vertex_program_metadata vertex_program_utils::analyse_vert
 	return result;
 }
 
+u64 RSXVertexProgram::get_sub_hash() const noexcept
+{
+	static constexpr u32 s_zero[4]{};
+
+	const usz prog_size = data.size() / 4;
+
+	const void* sdata_base = prog_size ? data.data() : s_zero;
+	const u8* sdata_start = static_cast<const u8*>(sdata_base);
+
+	const u8* sdata_mid1 = sdata_start + (prog_size / 8) * sizeof(v128);
+	const u8* sdata_mid2 = sdata_start + (prog_size / 4) * sizeof(v128);
```

Comment:
You're hashing this data without checking if they're constants or instructions first.